src/timeseriesUpdate.ts

```diff
@@ -73,3 +89,60 @@ export function updateNTokenPresentValueHistoricalData(notional: Notional, curre
   log.debug('Updated nTokenPresentValueHistoricalData variables for entity {}', [nTokenPresentValueHistoricalData.id]);
   nTokenPresentValueHistoricalData.save();
 }
+
+export function updateTvlHistoricalData(notional: Notional, maxCurrencyId: i32, timestamp: i32): void {
+  let tvlCurrencies = new Array<string>();
+  let usdTotal = BigInt.fromI32(0);
+
+  for (let currencyId: i32 = 1; currencyId <= 4; currencyId++) {
```

Comment:
This should iterate over max currencies

Reply:
Good catch, I added that for testing purposes.